test/utils.test.ts

```diff
@@ -971,8 +972,71 @@ describe("utils", () => {
 
             for (let i = 0; i < input.length; i++) {
                 utils.reportUrlType(input[i], reporter);
-                expect(reporter.sendTelemetryEvent).toBeCalledWith('user/browserNavigation', { 'urlType': expected[i] })
+                expect(reporter.sendTelemetryEvent).toBeCalledWith('user/browserNavigation', { 'urlType': expected[i] });
             }
         });
     });
+
+    describe('reportExtensionSettings', () => {
+        let mockVSCode: typeof import("vscode");
+
+        beforeEach(() => {
+            const vscodeMock = jest.requireMock("vscode");
+            vscodeMock.workspace.getConfiguration.mockImplementation(() => {
+                return {
+                    get: (name: string) => {
+                        switch(name) {
+                            case "enableNetwork":
+                                return true;
+                            case "themes":
+                                return "System preference";
+                            case "whatsNew":
+                                return true;
+                            default:
+                                return undefined;
+                        }
+                    },
+                    inspect: (name: string) => {
+                        switch(name) {
+                            case "enableNetwork":
+                                return {defaultValue: true};
+                            case "themes":
+                                return {defaultValue: "Light"};
+                            case "whatsNew":
+                                return {defaultValue: false};
+                            default:
+                                return {defaultValue: undefined};
+                        }
+                    },
+                    fillerPropertyOne: 'Need to add filler values since the extension related telemetry functions removes the first 4 entries to create an array of extension settings',
+                    fillerPropertyTwo: 'The first 4 entries for a workspaceConfiguration object are class related functions, the rest are all extension related settings.',
+                    themes: 'System preference',
+                    whatsNew: 'true',
+                    enableNetwork: 'false',
+                }
+            });

```

Comment:
I think we can save some repeating lines of code if we make a const with the getConfiguration object from helpers.ts and then use the spread operator. Like this:
```
const getConfig = vscodeMock.workspace.getConfiguration();
vscodeMock.workspace.getConfiguration.mockImplementation(() => {
     return {
          ...getConfig,
          ...other properties...
     }
});
```

Reply:
Good suggestion! Adding this change now.